source/docs/software/wpilib-tools/wpical/index.rst

```diff
@@ -0,0 +1,192 @@
+======
+WPIcal is a cross-platform utility that can be used to empirically measure the position and orientation of Apriltags. The primary use case for WPIcal is to make sure robots can perform as they would on an event field, but on a practice field that may have imperfect Apriltags setup.
+
+.. image:: images/WPIcal.png
+    :alt: WPIcal
+
+In Visual Studio Code, press :kbd:`Ctrl+Shift+P` and type ``WPILib`` or click the WPILib logo in the top right to launch the WPILib Command Palette. Select :guilabel:`Start Tool`, then select :guilabel:`WPIcal`.
+
+.. warning:: Before using this tool, it is important to understand that this tool is not a "silver bullet" of computer vision accuracy. The primary use case for WPIcal is to help detect and correct for incorrect Apriltag placement. While this tool can detect minor errors in precise tag positioning on a competition field, these corrections offer limited benefit and only for teams that have rigorously implemented all other good localization practices. With that said, this tool is exceedingly helpful in correcting for minor errors in practice fields, which are generally not within the normal field tolerance.
+
+Overview
+--------
+Conceptually, if you know how far tags are away from each other, and you know the field-relative coordinates of one of the tags, (the "pinned" tag) you can figure out the field-relative coordinates of the other tags. WPIcal achieves this by running an optimization on frames from videos that the user takes of the field. WPIcal is field-agnostic, meaning it will work with any FRC apriltag layout, as well as custom apriltag layouts, as long as they follow a couple of rules:
+
+* Translation components (in meters) are measured relative to the blue alliance origin.
+* Rotations are represented as Quaternions.
+
+Example:
+
+.. code-block:: json
+
+    {
+        "ID": 1,
+        "pose": {
+            "translation": {
+                "x": 15.079471999999997,
+                "y": 0.24587199999999998,
+                "z": 1.355852
+            },
+            "rotation": {
+                "quaternion": {
+                    "W": 0.5000000000000001,
+                    "X": 0.0,
+                    "Y": 0.0,
+                    "Z": 0.8660254037844386
+                }
+            },
+        },
+    }
+
+.. important:: WPIcal is meant to correct for SMALL variations in tag placement. It is still important that you set up your Apriltags in mostly the correct location and orientation, so WPIcal performs the optimal calibration.
+.. important:: Make sure that you verify the results of each calibration thoroughly to ensure that your calibration matches your field setup accurately.
+
+Camera Calibration
+------------------
+To measure the distance between tags in a video, you need the camera's intrinsics to account for things like distortion. WPIcal allows users to upload a video file to calibrate the camera in the tool, or upload a JSON file with your camera's intrinsics
```

Comment:
How should I capture a video file? I was able to do it on linux with gstreamer, but it was pretty non-trivial.
```
gst-launch-1.0 -vvv v4l2src device=/dev/video0 ! image/jpeg,format=MJPG,width=1600,height=1200,framerate=50/1 ! jpegdec ! videoconvert ! tee name=t t. ! queue ! autovideosink t. ! queue ! videoconvert ! avimux ! filesink location=output.avi
```

Reply:
I've been using a phone camera and then transferring the file to a computer to calibrate. For linux It will need to be transcoded. Do you think we should document how to capture the video for linux machines too?

Reply:
OBS also works well 

1. Add a V4L2 video capture device
2. Change properties:
![image](https://github.com/user-attachments/assets/39116f0d-ad0a-4f24-9e02-2ab04e2bbf66)
3. Under settings->video change the base resolution 
![image](https://github.com/user-attachments/assets/ac3c140e-9c5f-43b1-b27c-9b4ed1e35328)
4. Settings->output, not sure why i don't see avi on linux but it should be an option

Reply:
Do you think documenting the proper video capture method is within scope of the project? On all other platforms we're assuming users capture video beforehand.

Reply:
I think you should document all the steps a user'd have to take to get wpical's input files for all platforms